### PR TITLE
[Docs/Example] Adapt Escrow for Shared Object Deletion

### PR DESCRIPTION
**Pull request by amnn**
## Description

Adapt the Shared Escrow example and accompanying docs to account for the existence of Shared Object Deletion.

## Test Plan

```
examples/move/escrow$ sui move test
```

### CODE REVIEW

**Comment by amnn** [comment]
@giac-mysten and I were discussing how shared object deletion might change how we write Move code that interacts with shared objects, and I ended up writing the adapted version of Shared Escrow while we were discussing it.

Parking this here until Shared Object Deletion is fully landed (the tests will fail for now because Test Scenario does not like the fact that I took a shared object and then didn't give it back).

**Comment by tzakian** [comment]
Just FYI shared object deletion is on main, so the should(?) work now (unless there's something that wasn't done on the test scenario side that needs doing?)

**Comment by amnn** [comment]
Oh interesting, in that case, yes, there's something that needs to be done on the test scenario side!  It should now no longer be an error to take a shared object and then not return it.

**Comment by amnn** [comment]
It's possible though that the VM used for testing uses a ProtocolConfig that doesn't enable shared object deletion, even though it's landed in `main`?  Just speculating.

**Comment by tzakian** [comment]
Either way definitely worth checking out. I'll take a look and report back..

**Comment by tzakian** [comment]
@amnn Just put up a PR #14833 stacked on this one that adds the needed support for this. Happy to either merge it into this PR so we can land at the same time (and this can serve as a test for it), or can land it into main separately.

**Comment by amnn** [comment]
I don't mind either way, whatever's most convenient!

**Comment by amnn** [comment]
Changes look all good after your change to TestScenario @tzakian !  Now I think it's just a question of when shared object deletions becomes generally available to land this change to the example.

File: docs/content/guides/developer/sui-101/shared-owned.mdx [hunk 1]
```diff
@@ -140,32 +141,38 @@ This time the swap request is a shared object, and accepts the object being escr
...
+        id,
+        sender,
+        recipient,
+        exchange_key,
+        escrowed,
+    } = escrow;
 
-    let escrowed1 = option::extract<T>(&mut escrow.escrowed);
-    let escrowed2 = lock::unlock(locked, key);
+    assert!(recipient == tx_context::sender(ctx), EMismatchedSenderRecipient);
```
[the remark this answers is not in the document]
Comment (amnn) [reply docs/content/guides/developer/sui-101/shared-owned.mdx:157]: I mean this is a contrived example, so we can set whatever restrictions we like, but there are legitimate reasons to care about who you are trading with, and making sure you are only trading with that person. For example, if you cared about honouring sanctions, then you would want to know that the person you are trading with is not on the sanctions list.  Now, suppose that you initiate a trade with one person, and then that person transfers the `Key` to someone else who you didn't want to, or shouldn't be trading with -- now you're in trouble.

[the remark this answers is not in the document]
Comment (amnn) [reply docs/content/guides/developer/sui-101/shared-owned.mdx:157]: I would say the sanctions angle would be the reason to also encode the check that you are trading with who you thought you were trading with.

**Comment by amnn** [comment]
Superseded by #14833, #16197, #16198.

**Pull request closed without merging**